app/registries/foreman/plugin.rb

```diff
@@ -111,6 +115,18 @@ def registered_report_scanners
       def with_webpack
         all.select(&:uses_webpack?)
       end
+
+      def mark_finished!
+        @finished = true
+      end
+
+      def reset_finished!
+        @finished = false
+      end
+
+      def finished?
+        @finished
+      end
```

Comment:
This is probably nitpicking, but finished what? I'd include the verb which would render to something like `finished_initialization?`.

Reply:
called it `finished_registration`